stan/math/opencl/opencl_context.hpp

```diff
@@ -101,9 +101,12 @@ class opencl_context_base {
       if (device_properties & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE) {
         command_queue_ = cl::CommandQueue(
             context_, device_, CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, nullptr);
+        block_reads_ = CL_FALSE;
       } else {
         command_queue_ = cl::CommandQueue(context_, device_, 0, nullptr);
+        block_reads_ = CL_TRUE;
       }
+      block_writes_ = CL_TRUE;
```

Comment:
Why is this always true?

Reply:
I just wanted to see if the tests would pass, if always set to true. Then the only missing thing would be to only set it true for rvalues.